src/Elastic.Apm/Helpers/SystemInfoHelper.cs

```diff
@@ -161,27 +145,39 @@ protected virtual StreamReader GetCGroupAsStream()
 		internal const string PodUid = "KUBERNETES_POD_UID";
 		internal const string NodeName = "KUBERNETES_NODE_NAME";
 
-		internal KubernetesMetadata ParseKubernetesInfo(Container containerInfo, string hostName)
+		internal void ParseKubernetesInfo(Api.System system)
 		{
-			var @namespace = Environment.GetEnvironmentVariable(Namespace);
-			var podName = Environment.GetEnvironmentVariable(PodName);
-			var podUid = Environment.GetEnvironmentVariable(PodUid);
-			var nodeName = Environment.GetEnvironmentVariable(NodeName);
-
-			if (@namespace == null && podName == null && podUid == null && nodeName == null)
+			try
 			{
-				// By default, Kubernetes will set the hostname of the pod containers to the pod name.
-				// Users that override the name should use the Downward API to override the pod name.
-				return containerInfo != null
-					? new KubernetesMetadata { Pod = new Pod { Uid = containerInfo.Id, Name = hostName } }
-					: null;
-			}
+				var @namespace = Environment.GetEnvironmentVariable(Namespace);
+				var podName = Environment.GetEnvironmentVariable(PodName);
+				var podUid = Environment.GetEnvironmentVariable(PodUid);
+				var nodeName = Environment.GetEnvironmentVariable(NodeName);
+
+				if (@namespace != null || podName != null || podUid != null || nodeName != null)
```

Comment:
Maybe check with `string.IsNullOrEmpty` to also handle empty string.

Reply:
I considered this, but thought a user might want to set one of the values to an empty string, to overwrite a value parsed from /proc/self/cgroup. Happy to change, as treating an unset variable or variable with empty string the same is likely a better experience.